libcloud/storage/drivers/s3.py

```diff
@@ -1458,6 +1460,14 @@ class S3EUNorth1StorageDriver(S3StorageDriver):
     ex_location_name = "eu-north-1"
     region_name = "eu-north-1"
 
+class S3EUSouth1Connection(S3SignatureV4Connection):
```

Comment:
There is no need anymore to define a class per region. A long time ago we moved away from a class per region to `region` constructor argument approach. Other classes were just left in places for backward compatibility reasons.

Just adding an entry to `REGION_TO_HOST_MAP` dictionary is sufficient.

https://github.com/apache/libcloud/blob/9756484101d6570f03542288d2f38fdb381500d1/libcloud/storage/drivers/s3.py#L1253